### ISSUE TITLE
fbsimctl boots all simulators

### ISSUE THREAD

**Issue by bootstraponline** [facebook/FBSimulatorControl#152]
I'd expect boot to create one simulator that exactly matches the given name, not all of them. If I use the UUID then it works.

```
$ ./fbsimctl list
06139099-8108-4382-B9B3-3CC0ADB8E48E iPhone 4s iOS 9.2
...
```

```
$ ./fbsimctl boot "iPhone 4s iOS 9.2"
Booting 916F3B85-B553-4236-848C-03911CE65A9B iPhone 4s
Error Domain=com.facebook.FBSimulatorControl Code=0 "Expected Simulator 916F3B85-B553-4236-848C-03911CE65A9B to be Booted, but it was 'Booted'" UserInfo={launchd_is_running=0, launchd_subprocesses=[(null)], NSLocalizedDescription=Expected Simulator 916F3B85-B553-4236-848C-03911CE65A9B to be Booted, but it was 'Booted'}
medwards-l:build medwards$ ./fbsimctl boot "iPhone 4s iOS 9.2"
Booting 916F3B85-B553-4236-848C-03911CE65A9B iPhone 4s
Error Domain=com.facebook.FBSimulatorControl Code=0 "Expected Simulator 916F3B85-B553-4236-848C-03911CE65A9B to be Booted, but it was 'Booted'" UserInfo={launchd_is_running=0, launchd_subprocesses=[(null)], NSLocalizedDescription=Expected Simulator 916F3B85-B553-4236-848C-03911CE65A9B to be Booted, but it was 'Booted'}
medwards-l:build medwards$ ./fbsimctl boot "iPhone 4s iOS 9.2"
Booting 916F3B85-B553-4236-848C-03911CE65A9B iPhone 4s
Jan  6 00:07:46  FBSimulatorControl[18468] <Info>: 916F3B85-B553-4236-848C-03911CE65A9B: Did Change State => Booting
Jan  6 00:07:46  FBSimulatorControl[18468] <Info>: 916F3B85-B553-4236-848C-03911CE65A9B: Did Start => simulator_pid 18472 | launchd_sim_pid 18475
Jan  6 00:07:46  FBSimulatorControl[18468] <Info>: 916F3B85-B553-4236-848C-03911CE65A9B: Did Change State => Booted
Booted 916F3B85-B553-4236-848C-03911CE65A9B iPhone 4s
Booting 06139099-8108-4382-B9B3-3CC0ADB8E48E iPhone 4s iOS 9.2
Jan  6 00:07:46  FBSimulatorControl[18468] <Info>: 06139099-8108-4382-B9B3-3CC0ADB8E48E: Did Change State => Booting
Jan  6 00:07:46  FBSimulatorControl[18468] <Info>: 06139099-8108-4382-B9B3-3CC0ADB8E48E: Did Start => simulator_pid 18504 | launchd_sim_pid 18517
Jan  6 00:07:47  FBSimulatorControl[18468] <Info>: 06139099-8108-4382-B9B3-3CC0ADB8E48E: Did Change State => Booted
Booted 06139099-8108-4382-B9B3-3CC0ADB8E48E iPhone 4s iOS 9.2
Booting FAEA8059-5985-4F37-97A6-BEF0289E5618 iPhone 5
Jan  6 00:07:47  FBSimulatorControl[18468] <Info>: FAEA8059-5985-4F37-97A6-BEF0289E5618: Did Change State => Booting
Jan  6 00:07:48  FBSimulatorControl[18468] <Info>: FAEA8059-5985-4F37-97A6-BEF0289E5618: Did Start => simulator_pid 18547 | launchd_sim_pid 18557
Jan  6 00:07:48  FBSimulatorControl[18468] <Info>: FAEA8059-5985-4F37-97A6-BEF0289E5618: Did Change State => Booted
Booted FAEA8059-5985-4F37-97A6-BEF0289E5618 iPhone 5
Booting 279402FF-F382-42FE-913B-72C174508E67 iPhone 5
Jan  6 00:07:49  FBSimulatorControl[18468] <Info>: 279402FF-F382-42FE-913B-72C174508E67: Did Change State => Booting
Jan  6 00:07:50  FBSimulatorControl[18468] <Info>: 279402FF-F382-42FE-913B-72C174508E67: Did Start => simulator_pid 18603 | launchd_sim_pid 18617
Jan  6 00:07:50  FBSimulatorControl[18468] <Info>: 279402FF-F382-42FE-913B-72C174508E67: Did Change State => Booted
Booted 279402FF-F382-42FE-913B-72C174508E67 iPhone 5
Booting 9BEC0903-2A1E-4AAC-9D17-95AB6212FEBC iPhone 5s
Jan  6 00:07:53  FBSimulatorControl[18468] <Info>: 9BEC0903-2A1E-4AAC-9D17-95AB6212FEBC: Did Change State => Booting
Jan  6 00:07:53  FBSimulatorControl[18468] <Info>: 9BEC0903-2A1E-4AAC-9D17-95AB6212FEBC: Did Start => simulator_pid 18679 | launchd_sim_pid 18692
Jan  6 00:07:53  FBSimulatorControl[18468] <Info>: 9BEC0903-2A1E-4AAC-9D17-95AB6212FEBC: Did Change State => Booted
Booted 9BEC0903-2A1E-4AAC-9D17-95AB6212FEBC iPhone 5s
Booting A917A554-E007-463A-B6EE-A0C5EC9D43DC iPhone 6
Jan  6 00:07:57  FBSimulatorControl[18468] <Info>: A917A554-E007-463A-B6EE-A0C5EC9D43DC: Did Change State => Booting
Jan  6 00:07:57  FBSimulatorControl[18468] <Info>: A917A554-E007-463A-B6EE-A0C5EC9D43DC: Did Start => simulator_pid 18757 | launchd_sim_pid 18761
Jan  6 00:07:57  FBSimulatorControl[18468] <Info>: A917A554-E007-463A-B6EE-A0C5EC9D43DC: Did Change State => Booted
Booted A917A554-E007-463A-B6EE-A0C5EC9D43DC iPhone 6
Booting E3FF4EA6-2A34-4674-9A2F-85CC492C6E08 iPhone 6 Plus
...
```

**Comment by lawrencelomax** [comment]
This looks like erroneous behaviour to me! In future I want an empty `Query` to signify 'Perform the action with the last query that was run':

`boot 279402FF-F382-42FE-913B-72C174508E67` followed by `shutdown` would result in `279402FF-F382-42FE-913B-72C174508E67` being used as the query for `shutdown`.

When there isn't a 'prior simulator' it should probably bail out with a suitable error message. Could pick a simulator at random, but that's probably surprising behaviour.

**Comment by bootstraponline** [comment]
> When there isn't a 'prior simulator' it should probably bail out with a suitable error message.

:+1:

**Comment by lawrencelomax** [comment]
Will be fixed when #167 merges.
